components/footer/src/index.tsx

```diff
@@ -118,7 +118,7 @@ const FooterContainer = styled('footer')(
  * - https://github.com/alphagov/govuk-frontend/blob/main/src/govuk/components/footer/_footer.scss
  *
  */
-const Footer = ({ children, container: Container, copyright, meta, ...props }) => (
+const Footer = ({ children, container: Container = WidthContainer, copyright, meta, ...props }) => (
```

Comment:
weird one this - in principle as there's a `defaultProps` value set for `container` this shouldn't be needed, but the story shot default story fails to work properly without this

Reply:
this should be looked in to, the only thing I can think of right now is that the prop is

`container: Footer.WidthContainer`

perhaps make this `container: WidthContainer` so it still works if reordered?

Reply:
I think I'm happy with this PR as is other than this issue, and I want to run it through locally.

Reply:
it's odd - the `Page` container uses the same pattern, and the `defaultProps` value for `container` seems to work fine there

Reply:
this appears to be a bug with Storybook 6.3, there is a conversation on this here:

https://github.com/storybookjs/storybook/issues/15378#issuecomment-878968025

Reply:
Hopefully rolling back storyshots version to 6.2.9 will resolve this issue and this change won't be needed.

Also worth noting that defaultProps on functional components have been deprecated, so we'll need to deal with that at some point anyway, just I'd prefer not to do it as part of this PR.

Reply:
rolling back doesn't seem to fix the issue, will keep looking in to it

Reply:
ok so

- I think this is a bug in storybook 6.3, in that props can be set to undefined even if a default prop exists
- the same bug also exists in earlier versions of storyshots
- however, as default props on functional components are deprecated, I'm fine with us moving away from them in order to work round this bug

please can you remove all defaultProps in Footer and only use es6 defaults?

Reply:
although this doesn't explain why it's working on the Page component, will look in to that

Reply:
footer is exporting the component itself as the default story

`export const Default = Footer;`

however Page is wrapping this

```
export const Default = () => (
  <div>
    <Page beforeChildren={<BackLink href="#" />}>
      <H1>Page Title</H1>
    </Page>
  </div>
);
```

which is likely the reason it's working for Page and not Footer

If we change the default Page story to `export const Default = Page;` then we get the same issue.

Reply:
ok cool - I'll remove the `defaultProps` from footer - I wasn't really aware that they were deprecated, but I do tend to prefer using es6 defaults, especially in typescript

medium term we should probably look to remove `propTypes` too and use TS types instead - but that's for another time

Reply:
Yeah it had snuck me by too https://github.com/facebook/react/pull/16210

Reply:
there's a new note that's been added to the storybook bug conversation showing how this can be fixed by changing the default story for the Footer, so I've now restored the `defaultProps`, removed this es6 default, and changed the story

if/when we move from `propTypes` to TS type declarations then we can put the es6-style value default back in